src/ChessBoard.scala

```diff
@@ -0,0 +1,153 @@
+import com.sun.org.apache.xpath.internal.operations.Or
+
+object ChessBoard extends App {
+
+  type Board = Seq[Seq[String]]
+
+  // Takes a number - say 4 (n)
+  // Uses this to create a board which is  4 x 4 (n x n)
+  def printBoard(board: Board): Unit = {
+    // mkString takes elements in a List and combines them with the specified characters
+    // ex. Seq(1, 2, 3).mkString(" ~ ") would output: "1 ~ 2 ~ 3"
+    val boardText = board.map(row => row.mkString(", ")).mkString("\n")
+    println(boardText)
+  }
+
+  def createBoard(n: Int): Board = {
+    // Empty board
+    // Figure out how to create board
+    val board = Seq.fill(n)(Seq.fill(n)("-"))
+    // try to populate first row
+    // Make Pusdo code comments
+    //printSolution(board)
+    board
+  }
+
+  def markXs(board: Board, row: Int, col: Int): Board = {
+
+    val newRow = board(row).updated(col, "X")
```

Comment:
Instead of using the strings "-", "Q", "X" in your logic, you could consider doing something like:
```
sealed trait Square { val value: String }

object Square {
  case object Empty extends Square { val value = "-" }
  case object Queen extends Square { val value = "Q" }
  case object Invalid extends Square { val value = "X" }

  def fromValue(value: String): Square = value match {
    case "-" => Empty
    case "Q" => Queen
    case "X" => Invalid
  }
}
```
Then in your code instead of referencing `"Q"`  you could do `Queen`

This is a good, simple example of Scala pattern matching and traits